vtr_flow/scripts/slurm/submission_template.sh

```diff
@@ -0,0 +1,27 @@
+#! /bin/bash
+
+#SBATCH --nodes=1                                       # Number of nodes in our pool
+#SBATCH --ntasks=80                                     # Number of parallel tasks
+#SBATCH --ntasks-per-node=40                            # maximum number of tasks simultaneously running on a node
+#SBATCH --job-name=vtr_test                             # Job name
+#SBATCH --mail-type=FAIL,END                            # When to send an email
+#SBATCH --mail-user=XXXX@mail.utoronto.ca               # The user's email
```

Comment:
Is there anyway to template these SBATCH directives with environment variables?

Reply:
Sure, we can! 
Just by replacing all of them with
`#SBATCH --export=ALL`

and let the user set the following environment variables:
- SLURM_JOB_NUM_NODES instead of --nodes
- SLURM_NTASKS or SLURM_NPROCS instead of --ntasks
- SLURM_NTASKS_PER_NODE instead of --ntasks-per-node
- SLURM_JOB_ID instead of %j
- SBATCH_JOB_NAME instead of --job-name

However, the question is which one will be easier to use .. the environment variables or the script local ones?

Reply:
I think editing the file is somewhat more difficult for 2 reasons, but I would understand if you disagree

1. By editing the file users now need to remove any local changes for their clusters before submitting PRs to the repo. It's also more likely that a subtle change will slip through the cracks into the repo.
2. When deploying to different cloud environments you often want to have the ability to change the configuration depending on the environment. Doing this through env variables is easier than doing it through file modifications. 

To me I think a good compromise would be to use the environment mode, but include a comment with the most common/required fields.

Reply:
Makes sense. I tested it and unfortunately it seems that most of these environment variables are output from SLURM based on the allocated nodes/tasks/tasks-per-node. The only input environment variables I can found are:
- SBATCH_JOB_NAME instead of --job-name
- SBATCH_TIMELIMIT instead of --time

For all the other parameters, I am not able to find an env. variable to overwrite them. Hence, I have added more commenting in the script explaining these options.

Reply:
Interesting, this probably means in the long run we would have to dynamically generate a bash file in order to template slurm jobs. This change seems good for now given the current landscape, but I would love to see this become more dynamically configurable.